app/src/main/java/com/aileyzhang/musicapp/adapters/SongListAdapter.java

```diff
@@ -31,8 +32,9 @@ public SongListAdapter(@NonNull Context context, int resource, @NonNull List<Son
     public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
         View songListView = convertView;
         if (songListView == null) {
-            songListView = LayoutInflater.from(getContext()).inflate(R.layout.song_list_item,
-                    parent, false);
+//            songListView = LayoutInflater.from(getContext()).inflate(R.layout.song_list_item,
+//                    parent, false);
+            songListView = new SongListItemView(getContext(), parent);
```

Comment:
Don't pass the parent here, instead instantiate your custom view here and simply return it.